source/robot_model/src/Model.cpp

```diff
@@ -29,17 +39,175 @@ bool Model::create_urdf_from_string(const std::string& urdf_string, const std::s
   return false;
 }
 
+std::set<std::string> Model::extract_package_name_from_urdf() {
+    std::set<std::string> extracted_words;
+    std::regex package_path_pattern(R"(filename=\"([^\"]+)\")");
+    std::regex package_name_pattern(R"(package://([^/]+)/)");
+
+    std::smatch matches_file;
+    std::smatch matches_package_name;
+    auto start = this->urdf_.str().cbegin();
+    auto end = this->urdf_.str().cend();
+
+    // Extract package paths
+    while (std::regex_search(start, end, matches_file, package_path_pattern)) {
+        std::string path = matches_file[1];
+        
+        if (std::regex_search(path, matches_package_name, package_name_pattern)) {
+            extracted_words.insert(matches_package_name[1]);
+        }
+        
+        start = matches_file[0].second;   
+    }
+    return extracted_words;
+}
+
+void Model::replace_package_by_full_path(const std::string& package_name, const std::string& full_path) {
+    std::string target = "package://" + package_name + "/";
+    std::string replacement = full_path;
+
+    std::string urdf = this->urdf_.str();
+
+    // Start from the beginning of the result string
+    size_t start_position = 0;
+    while ((start_position = urdf.find(target, start_position)) != std::string::npos) {
+        // Replace the target with the replacement string
+        urdf.replace(start_position, target.length(), replacement);
+        // Move past the last replacement
+        start_position += replacement.length();
+    }
+    this->urdf_.str(urdf);
+}
+
+void Model::resolve_package_paths() {
+    std::set<std::string> package_names = this->extract_package_name_from_urdf();
+
+    for (const auto& package_name : package_names) {
+        std::string package_path = this->meshloader_callback_(package_name);
+        this->package_paths_.push_back(package_path);
+        this->replace_package_by_full_path(package_name, package_path);
+    }
+}
+
+// read the URDF file and return the string
```

Comment:
This kind of comment is not necessary if you have written a good docstring in the header. Duplicated documentation can be more confusing than anything else (just like DRY for code)